### ISSUE TITLE
QuickContainers cmake deprecation warning

### ISSUE THREAD

**Issue by petrmanek** [cneben/QuickQanava#221]
I have been getting the following cmake deprecation warning when configuring the project using any remotely recent cmake version (currently running v3.26):

```
CMake Deprecation Warning at lib/quickqanava/QuickContainers/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
```

I suspect this is due to an [old minimum version](https://github.com/cneben/QuickQanava/blob/004d3e42f5e90b5916e98edf5da7bda5a5525d07/src/quickcontainers/CMakeLists.txt#L1) declared in the respective CMakeLists.txt. Quickly glancing at the contents of the file, I do not see anything preventing this version from being bumped to at least 3.5.

**Comment by cneben** [comment]
Yes, unfortunately, there is a minimal version mismatch across the project, I've just switched to `cmake_minimum_required(VERSION 3.16)` which is Qt6 minimum in develop (soon 2.5.0) branch.

I will try to create a 2.4.1 branch from 2.4.0 tag this afternoon and fix this problem + #220 

Are you still interested in Qt5 branch ? Otherwise you should already switch to develop it is already in our production pure Qt6 version (still need some polish for node dragging, but definititely useable).

**Comment by petrmanek** [comment]
Awesome, 2.4.1 appears to solve this. Thanks for expeditious reply!


> Are you still interested in Qt5 branch ? Otherwise you should already switch to develop it is already in our production pure Qt6 version (still need some polish for node dragging, but definititely useable).

No, I am not -- I am also running in pure Qt6 mode for a while now.
